lib/pbench/server/api/resources/query_apis/datasets/dataset_mappings.py

```diff
@@ -95,11 +95,11 @@ def _get(self, index_key: AnyStr, request: Request) -> Dict[AnyStr, Any]:
         # don't go through JSON schema validation, so we have
         # to do it here.
         try:
-            self.schema.validate(index_key)
+            self.schema.validate(dataset_view)
         except SchemaError as e:
             abort(HTTPStatus.BAD_REQUEST, message=str(e))
 
-        index = RunIdBase.ES_INTERNAL_INDEX_NAMES[index_key["index_key"]]
+        index = RunIdBase.ES_INTERNAL_INDEX_NAMES[dataset_view["dataset_view"]]
```

Comment:
I'm confused...what is going on in this statement?  According to the type hint at line 84, `dataset_view` is a string...but from the access, it looks like it's a dictionary.  (And, having the name of the dict and the key value be the same looks quite weird to me.)

Perhaps the dict should be renamed to "url_parameters" or similar?  (And, the type hint should be corrected!)

Reply:
Yeah The type hint needs to be updated, it is a dict